packages/tailwindcss/src/variants.ts

```diff
@@ -782,7 +894,7 @@ export function createVariants(theme: Theme): Variants {
 
               ruleNode.nodes = [rule(`@media (width < ${value})`, ruleNode.nodes)]
             },
-            { compounds: false },
+            { compounds: ['at-rule'] },
```

Comment:
`compounds` here means that this rule _might_ contribute an _at-rule_ or a style rule. This wasn't super clear to me from the name, perhaps we can find a better alternative? It feels like the relationship from the compounding rule to the "simple" one is "compounds" but not sure what to call it in the other direction. 

Something like `isValidForCompoundsThatWorkWith` would explain the behavior, let's think about some name in that direction.

Reply:
After much discussion we're just going to leave it as is for now because its an internal name and we couldn't come up with anything good.